src/sentry/api/endpoints/organization_code_mappings.py

```diff
@@ -0,0 +1,199 @@
+from django.http import Http404
+from django.utils.translation import ugettext_lazy as _
+from rest_framework import serializers, status
+
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationIntegrationsPermission
+from sentry.api.serializers import serialize
+from sentry.api.serializers.rest_framework.base import CamelSnakeModelSerializer
+from sentry.models import OrganizationIntegration, Project, Repository, RepositoryProjectPathConfig
+from sentry.utils.compat import map
+
+
+def gen_path_regex_field():
+    return serializers.RegexField(
+        r"^[^\s'\"]+$",  # may need to add more characters to prevent in the future
+        required=True,
+        allow_blank=True,
+        error_messages={"invalid": _("Path may not contain spaces or quotations")},
+    )
+
+
+class RepositoryProjectPathConfigSerializer(CamelSnakeModelSerializer):
+    repository_id = serializers.IntegerField(required=True)
+    project_id = serializers.IntegerField(required=True)
+    stack_root = gen_path_regex_field()
+    source_root = gen_path_regex_field()
+    default_branch = serializers.RegexField(
+        r"^(^(?![\/]))([\w\/-]+)(?<![\/])$",
+        required=True,
+        error_messages={
+            "invalid": _(
+                "Branch name may only have letters, numbers, underscores, forward slashes and dashes. Branch name may not start or end with a forward slash."
+            )
+        },
+    )
+
+    class Meta:
+        model = RepositoryProjectPathConfig
```

Comment:
@MeredithAnya and the model name?

Reply:
@NisanthanNanthakumar I don't know that we need to go through the trouble of renaming the model (or serializer), I just wanted to make sure that external facing code (the endpoint routes) we use has the code mappings terminology

Reply:
@MeredithAnya hmm I'm of the opinion that we should keep the naming consistent and try to reflect its use even internally. Although changing the tablename cannot justify the effort required, changing the names elsewhere should be relatively trivial.

Reply:
@NisanthanNanthakumar because the serializer is closely tied to the model, and I don't think its worth changing the name of the model, I think it's less confusing to have the serializer name match the model name.